app/controllers/admin/base_controller.rb

```diff
@@ -0,0 +1,3 @@
+class Admin::BaseController < ApplicationController
+  before_action :admin_gate
```

Comment:
@s-espinosa @carmer

Made a base controller to handle admin requests to admin controllers.